packages/firestore/src/remote/remote_store.ts

```diff
@@ -242,6 +237,9 @@ export class RemoteStore {
    * enableNetwork().
    */
   disableNetwork(): Promise<void> {
+    if (!this.isNetworkEnabled()) {
+      return Promise.resolve();
+    }
```

Comment:
[The beauty of multiple ports is you get to think about the change multiple times and catch bugs in your 2nd or 3rd review... :-)]

Currently disableNetworkInternal() is not idempotent (in particular, it'll crash on subsequent calls). And we have 2 places we call disableNetworkInternal() besides here, and I think both are problematic.  shutdown() should not call disableNetworkInternal() if it's already disabled, and handleUserChange() should not call disableNetworkInternal() /or/ enableNetwork() if it's currently disabled.

Reply:
Ok, I think I've addressed this now. In one case I've added an assert, which I think is safe, but take a look.